File: scripts/generate_shadows.mjs
import sharp from 'sharp';
import path from 'path';
import fs from 'fs';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const BLUR_SIGMA = 1.5;

const birds = [
  'cedar_waxwing.png',
  'cardinal.png',
  'robin.png',
  'hummingbird.png',
  'painted_bunting.png',
  'house_sparrow.png',
  'scissor_tailed_flycatcher.png',
  'blue_jay.png',
];

const assetsDir   = path.join(__dirname, '..', 'assets', 'birds');
const shadowsDir  = path.join(assetsDir, 'shadows');
if (!fs.existsSync(shadowsDir)) fs.mkdirSync(shadowsDir);

for (const filename of birds) {
  const src = path.join(assetsDir, filename);
  if (!fs.existsSync(src)) {
    console.log(`  skip ${filename} (not found)`);
    continue;
  }

  const img = sharp(src);
  const { width, height } = await img.metadata();
  const { data } = await img.ensureAlpha().raw().toBuffer({ resolveWithObject: true });

  // Zero RGB channels, keep alpha — produces a black silhouette
  for (let i = 0; i < data.length; i += 4) {
    data[i] = 0; data[i + 1] = 0; data[i + 2] = 0;
  }

  const stem = filename.replace(/\.png$/i, '');
  const out = path.join(shadowsDir, `${stem}_shadow.png`);
  await sharp(Buffer.from(data), { raw: { width, height, channels: 4 } })
    .blur(BLUR_SIGMA)
    .png()
    .toFile(out);

  console.log(`  wrote ${path.basename(out)}`);
}

console.log('Done.');

File: scripts/generate_shadows.py
#!/usr/bin/env python3
"""
Generate silhouette drop-shadow PNGs for each bird asset.
Run once from the project root: python scripts/generate_shadows.py
Requires: pip install Pillow
"""
from PIL import Image, ImageFilter
import os

BLUR_RADIUS = 12

birds = [
    ('waxwing',        'cedar_waxwing.png'),
    ('cardinal',       'cardinal.png'),
    ('robin',          'robin.png'),
    ('hummingbird',    'hummingbird.png'),
    ('paintedbunting', 'painted_bunting.png'),
]

assets_dir = os.path.join(os.path.dirname(__file__), '..', 'assets', 'birds')

for bird_id, filename in birds:
    src = os.path.join(assets_dir, filename)
    if not os.path.exists(src):
        print(f'  skip {filename} (not found)')
        continue

    img = Image.open(src).convert('RGBA')
    *_, alpha = img.split()

    shadow = Image.merge('RGBA', [
        Image.new('L', img.size, 0),
        Image.new('L', img.size, 0),
        Image.new('L', img.size, 0),
        alpha,
    ]).filter(ImageFilter.GaussianBlur(radius=BLUR_RADIUS))

    out = os.path.join(assets_dir, f'{bird_id}_shadow.png')
    shadow.save(out)
    print(f'  wrote {os.path.basename(out)}')

print('Done.')
